rest/route/service.go

```diff
@@ -241,6 +241,7 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/tasks/{task_id}/tests").Version(2).Get().Wrap(addProject, viewTasks).RouteHandler(makeFetchTestsForTask(env, sc))
 	app.AddRoute("/tasks/{task_id}/tests/count").Version(2).Get().Wrap(addProject, viewTasks).RouteHandler(makeFetchTestCountForTask())
 	app.AddRoute("/tasks/{task_id}/sync_path").Version(2).Get().Wrap(requireUser).RouteHandler(makeTaskSyncPathGetHandler())
+	app.AddRoute("/tasks/{task_id}/generated_tasks").Version(2).Get().Wrap(requireUser, viewTasks).RouteHandler(makeGetGeneratedTasks())
```

Comment:
Rather than creating a new route, did we consider adding this as an optional parameter to the existing get task route?

Reply:
I was thinking it could be part of the APITask struct and just conditionally populated, similar to how we have the APITaskArgs here https://github.com/evergreen-ci/evergreen/blob/0e4178aa0c4bd3b7690a351fca870efcb31bf41a/rest/model/task.go#L369 
I mostly ask this because having optional parameters is pretty consistent with a lot of behavior, and could be less bespoke. I'm fine with whichever though since this works and is kind of its own thing, just wanted to mention it.

Reply:
I guess we could return an APITask but it would be kind of awkward still because you would end up with a long list of JSON docs with most of the fields empty, which would make an argument for "why not just return the entire task". Anyways, I was hoping to keep this route's results simple. Since this returns a strict subset of the fields in a task, we can always adapt it later if need arises to return all the results + change the route name.